libs/vm-modules/tests/unit/ml_model_tests.cpp

```diff
@@ -347,6 +347,140 @@ TEST_F(VMModelTests, model_init_with_wrong_name)
   EXPECT_THROW(toolkit.Run(), std::runtime_error);
 }
 
+TEST_F(VMModelTests, model_add_invalid_layers)
```

Comment:
this test has nice coverage, but it would be preferable to wrap up the re-usable code in the test fixture, and then have multiple small tests; that way when one fails you immediately know which invalid case is at fault

Reply:
Agreed